modules/apps/object/object-web/src/main/resources/META-INF/resources/js/utils/getNonInheritanceObjectRelationshipFields.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {API} from '@liferay/object-js-components-web';
+
+const getInheritanceObjectRelationshipExternalReferenceCodes = async (
+	objectRelationshipExternalReferenceCodes: string[],
+	parentObjectDefinitionExternalReferenceCode: string
+): Promise<string[]> => {
+	if (objectRelationshipExternalReferenceCodes.length === 1) {
+		return objectRelationshipExternalReferenceCodes;
+	}
+
+	try {
+		const objectRelationships =
+			await API.getObjectDefinitionByExternalReferenceCodeObjectRelationships(
+				parentObjectDefinitionExternalReferenceCode,
+				{
+					filter: 'edge eq true',
+				}
+			);
+
+		return objectRelationships.map(
+			({externalReferenceCode}) => externalReferenceCode
+		);
+	}
+	catch (error) {
+		console.error(
+			`Failed to fetch inheritance relationships for parent ${parentObjectDefinitionExternalReferenceCode}:`,
+			error
+		);
+
+		return [];
+	}
+};
+
+const groupObjectRelationshipsByParent = (
+	objectRelationshipFields: ObjectField[]
+): Map<string, string[]> => {
+	const objectRelationshipsByParent = new Map<string, string[]>();
+
+	for (const {
+		objectDefinitionExternalReferenceCode1,
+		objectRelationshipExternalReferenceCode,
+	} of objectRelationshipFields) {
+		if (
+			objectDefinitionExternalReferenceCode1 &&
+			objectRelationshipExternalReferenceCode
+		) {
+			const existing =
+				objectRelationshipsByParent.get(
+					objectDefinitionExternalReferenceCode1
+				) ?? [];
+
+			objectRelationshipsByParent.set(
+				objectDefinitionExternalReferenceCode1,
+				[...existing, objectRelationshipExternalReferenceCode]
+			);
+		}
+	}
+
+	return objectRelationshipsByParent;
+};
+
+export async function getNonInheritanceObjectRelationshipFields(
+	objectDefinition: ObjectDefinition
+): Promise<ObjectField[]> {
+	const {objectDefinitionSettings, objectFields} = objectDefinition;
+
+	if (!objectDefinitionSettings) {
+		return objectFields;
+	}
+
+	const objectDefinitionSettingsMap = new Map(
+		objectDefinitionSettings.map((objectDefinitionSetting) => [
+			objectDefinitionSetting.name,
+			objectDefinitionSetting.value,
+		])
+	);
+
+	const rootObjectDefinitionExternalReferenceCodes =
+		objectDefinitionSettingsMap.get(
```

Comment:
this is more a question than asking for a change: did you create `objectDefinitionSettingsMap` with the intention to use it in the future for other settings?

Reply:
Hi @rebsilva, good question. Perhaps, when I was creating this function, using Map made more sense. But we can safely use find. In the worst-case scenario, both have O(n) time complexity. But we use less code with the `find.()` approach.